### ISSUE TITLE
QuestionHelper autocomplete with trailing backslashes triggers display error

### ISSUE THREAD

**Issue by ciaranmcnulty** [symfony/symfony#24652]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.3.10

Providing trailing backslashes in an autocomplete value seems to break the console formatting, when using QuestionHelper. I assume this is a result of faulty escaping around the highlighting:

```php
$questionHelper = $this->getApplication()->getHelperSet()->get('question');
$question = new Question('Enter namespace:');
$question->setAutocompleterValues(['Foo\\', 'Bar\\']);
$namespace = $questionHelper->ask($input, $output, $question);
```

<img width="202" alt="screen shot 2017-10-21 at 08 40 41" src="https://user-images.githubusercontent.com/237866/31849558-220ae288-b63c-11e7-829e-cf2edf5682c7.png">

**Comment by ciaranmcnulty** [comment]
Additional info from some playing with the prompt.

Initial prompt looks good:
`Enter namespace:`

Typing a match causes the broken display:
`Enter namespace:F[cursor here]oo\</hl>`

Hitting tab to complete actually completes to the wrong value (too many `\`):
`Enter namespace:Foo\\[cursor here]/hl>`

**Comment by chalasr** [comment]
Fixed in #24660
